dev/core/src/com/google/gwt/dev/jjs/impl/CloneExpressionVisitor.java

```diff
@@ -255,6 +256,11 @@ public void endVisit(JPermutationDependentValue x, Context ctx) {
         "this point but contains " + x);
   }
 
+  @Override
+  public boolean visit(JSwitchExpression x, Context ctx) {
```

Comment:
Really intentional. 

Consider that the switch expression is the _only_ kind of java expression that includes new statements which exist in the same scope as the statement in which the expression itself lives. As a result, if you include switch-expressions in the cloner, suddenly the cloner must be a general purpose Node cloner, since any statement can be cloned as well.